### PR TITLE
Enhance PtMobile type

### PR DESCRIPTION
**Pull request by danielsp45**
Enhancing the PtMobile type, while taking advantage of the [ex_phone_number](https://github.com/ex-phone-number/ex_phone_number) library which allows for a more complete validation.

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -93,7 +93,8 @@ defmodule Bokken.MixProject do
       # tools
       {:credo, "~> 1.6", only: [:dev, :test], runtime: false},
       {:dialyxir, "~> 1.1", only: [:dev, :test], runtime: false},
-      {:ex_doc, "~> 0.28", only: [:dev], runtime: false}
+      {:ex_doc, "~> 0.28", only: [:dev], runtime: false},
+      {:ex_phone_number, "~> 0.3"}
```
Comment (nelsonmestevao) [inline mix.exs]: move this to utilities :eyes:

File: lib/bokken/ecto/pt_mobile.ex [hunk 2]
```diff
@@ -13,66 +13,46 @@ defmodule Bokken.Ecto.PtMobile do
...
+  defp format(number) do
+    {:ok, phone_number} = ExPhoneNumber.parse(number, "PT")
 
-    if v >= 48 and v <= 57 do
-      [n | format(rest)]
+    if ExPhoneNumber.is_valid_number?(phone_number) do
+      {:ok, ExPhoneNumber.format(phone_number, :e164)}
     else
-      format(rest)
+      {:error, [message: "invalid PT phone number"]}
```
Comment (nelsonmestevao) [inline lib/bokken/ecto/pt_mobile.ex]:
```suggestion
      {:error, [message: gettext("invalid PT phone number")]}
```

So we keep the possibility to have translations :pray:

**Review by ruilopesm — COMMENTED** [review]
Since the tests module is very small now, _doctesting_ would be a good way to go 👀

**Comment by danielsp45** [comment]
> Since the tests module is very small now, _doctesting_ would be a good way to go 👀

Just implemented it. It makes the module nicely documented 👌

**Review by ruilopesm — APPROVED** [review]
Great work! 🙌🏼

File: lib/bokken/ecto/pt_mobile.ex [hunk 3]
```diff
@@ -1,7 +1,27 @@
...
+
+    iex> Bokken.Ecto.PtMobile.cast("+351 979 066 855")
+    {:error, [message: "invalid PT phone number"]}
+
+    iex> Bokken.Ecto.PtMobile.cast("+351 989 066 855")
+    {:error, [message: "invalid PT phone number"]}
   """
+  import BokkenWeb.Gettext
 
   use Ecto.Type
```
Comment (nelsonmestevao) [inline lib/bokken/ecto/pt_mobile.ex:26]: You should keep use first, tacking into account it's special nature.

3rd point on CeSIUM's guides for Elixir: https://github.com/cesium/guides/tree/main/elixir#elixir

@danielsp45 @RuiL1904 @ruioliveira02 (I think this is the purpose of a style guide).

**Pull request merged**